presto-main/src/main/java/io/prestosql/execution/QueryState.java

```diff
@@ -28,6 +28,10 @@
      * Query is waiting for the required resources (beta).
      */
     WAITING_FOR_RESOURCES(false),
+    /**
```

Comment:
Commit message:

`Add DISPATCHING query state`

Would also recommend some more detail about what this state means in the full message.